app/views/shared/_services-nav.html.erb

```diff
@@ -1,50 +1,108 @@
-            <% else %>
-                <span class="shortcuts__link"><%= link_to directory["value"], admin_services_path(directory: directory["value"])%></span>
-            <% end %>
-            <span class="shortcuts__count">(<%= @directories_counts[directory["name"]] %>)</span>
-            <span class="shortcuts__divider">|</span>
-        <% end %>
-    <% end %>
-
-    <% if (params[:ofsted] === "true") %>
-        <span class="shortcuts__current">Ofsted registered</span>
-    <% else %>
-        <%= link_to "Ofsted registered", admin_services_path(ofsted: true), class: "shortcuts__link" %>
-    <% end %>
-
-    <span class="shortcuts__count">(<%= @ofsted_count %>)</span>
-
-    <span class="shortcuts__divider">|</span>
-
-    <%= link_to_unless_current "Pending", admin_requests_path, class: "shortcuts__link" do %>
-        <span class="shortcuts__current">Pending</span>
-    <% end %>
-    <span class="shortcuts__count">(<%= @pending_count %>)</span>
-
-    <span class="shortcuts__divider">|</span>
-
-    <% if (params[:archived] === "true") %>
-        <span class="shortcuts__current">Archived</span>
-    <% else %>
-        <%= link_to "Archived", admin_services_path(archived: true), class: "shortcuts__link" %>
-    <% end %>
-
-    <span class="shortcuts__count">(<%= @archived_count %>)</span>
+  <% 
+    @directory_links = {}
+
+    if APP_CONFIG["directories"].present?
+      APP_CONFIG["directories"].each do |directory|
+        @directory_links[directory["value"]] = admin_services_path(directory: directory["value"]);
+      end
+    end
+
+%>
+  
+  <%= 
+  # display the bfis/bod headings
+  render partial: "shared/directory-shortcuts",  locals: { 
+        all_state: !params[:directory].present?,
+        all_link: admin_services_path,
+        all_count: @service_counts[:all][:all],
+        all_label: "All",
+        directory_links: @directory_links,
+        directory_counts: @service_counts
+  }
+
+  #directory_link: admin_services_path(directory: directory["value"])
+%>
+
+
+
+<nav class="shortcuts shortcuts--no-top-spacing  shortcuts--sublevel">
+
+<% 
+
+
+    @sublevel_ofsted_state = false;
+    @sublevel_pending_state = false;
+    @sublevel_archived_state = false;
+
+    # toggle the active state for the ofsted filter
+    if params[:ofsted].present? && params[:ofsted] === "true" 
+        @sublevel_ofsted_state = true;
+    else    
+        @sublevel_ofsted_state = false;
+    end
+
+    # toggle the active state for the pending filter
+    if current_page?(controller: 'admin/requests', action: 'index') 
+        @sublevel_pending_state = true;
+    else    
+        @sublevel_pending_state = false;
+    end
+
+    # toggle the active state for the archived filter
+    if params[:archived].present? && params[:archived] === "true" 
+        @sublevel_archived_state = true;
+    else    
+        @sublevel_archived_state = false;
+    end
+    
+
```

Comment:
I went back and forth on this - originally (before the partials) the logic was much more complex but now its much simpler so I've refactored it but kept it in the views. 

I was going to move it up into the controller as suggested but if I moved it into the controller there would be duplicate logic in the requests controller too (for the pending) plus I do still kind of feel that the logic more directly relates to the view rather than the controller. Since it deals with whats visible, what the link will be and fetching the count which is already set in the controller we're just fetching different bits